### PR TITLE
[NEW] Add steps deadlines informaion

### PR DESCRIPTION
**Pull request by eberloso**
Adds the data about the maximum days you have to answer for each step.

### CODE REVIEW

**Comment by ecarreras** [comment]
I propose you to rewrite this into self explained code, without hard to read structures and conventions like 'DN' or 'DA'. Think something like:
```python
from collections import namedtuple
from datetime import datetime, timedelta
from workdays import workday

from expects import *

class DeadLine(namedtuple('Deadline', "step, days, next_step")):
    def limit(self, date):
        if isinstance(self.days, Naturaldays):
            return date + timedelta(self.days)
        else:
            return workday(date, self.days)

class Workdays(int):
    def __new__(cls, *args, **kwargs):
        return super(Workdays, cls).__new__(cls, *args, **kwargs)


class Naturaldays(int):
    def __new__(cls, *args, **kwargs):
        return super(Naturaldays, cls).__new__(cls, *args, **kwargs)


class ProcessDeadline(object):

    @classmethod
    def get_deadline(cls, step):
        for s in cls.steps:
            if s.step == step:
                return s


with description('A Dead Line'):
    with it('must have and object representing the deadline'):

        dl = DeadLine('01', Naturaldays(5), '02')
        expect(dl).to(have_properties(step='01', days=5, next_step='02'))

    with it('must have to calculate the natual days of the deadline'):
        dl = DeadLine('01', Naturaldays(5), '02')
        limit = dl.limit(datetime(2016, 12, 1))
        expect(limit).to(equal(datetime(2016, 12, 6)))

    with it('must have to calculate the working days of the deadline'):
        dl = DeadLine('01', Workdays(5), '02')
        limit = dl.limit(datetime(2016, 12, 1))
        expect(limit).to(equal(datetime(2016, 12, 8)))

with description('A Process deadline'):
    with it('must permit get the limit by step'):

        class C2(ProcessDeadline):
            steps = [
                DeadLine('01', Workdays(5), '02'),
                DeadLine('02', Workdays(1), '05'),
                DeadLine('08', Workdays(5), '09')
            ]

        class R1(ProcessDeadline):
            steps = [
                DeadLine('01', Workdays(5), '02'),
                DeadLine('03', Naturaldays(20), '04')
            ]

        c2 = C2()
        r1 = R1()

        c2_dl = c2.get_deadline('01')
        expect(c2_dl).to(equal(DeadLine('01', Workdays(5), '02')))

        r1_dl = r1.get_deadline('03')
        expect(r1_dl).to(equal(DeadLine('03', Naturaldays(20), '04')))
```

What we get with this:

- The logic remains here, not in the ERP code.
- We can add the `steps` into each message class like `swiching.input.messages.C1`.
- Better testing decoupling, all the logic must be here.

File: switching/input/messages/Deadlines.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 from collections import namedtuple
-from datetime import timedelta
+from datetime import datetime, timedelta
 from workdays import workday
 
 
 class DeadLine(namedtuple('Deadline', "step, days, next_step")):
-    def limit(self, date):
+    def limit(self, date=datetime.today()):
```
Comment (ecarreras) [inline switching/input/messages/Deadlines.py]: It's better to initialize with `None` and then check `if date is None` assign `datetime.today()`

File: switching/input/messages/Deadlines.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from collections import namedtuple
+from datetime import datetime, timedelta
+from workdays import workday
+
+
+class DeadLine(namedtuple('Deadline', "step, days, next_step")):
```
Comment (guilleJB) [inline switching/input/messages/Deadlines.py]: La namedtuple té 4 parametres o són 2 ja que hi ha unes " que comencen al step i es tanquen al final del next_step ?

Comment (ecarreras) [inline switching/input/messages/Deadlines.py]: Té 3 paràmetres, el namedtuple es poden definir de les dues formes:
```python
NamedOne = namedtuple('NamedOne', ['one', 'two', 'three'])
NamedTwo = namedtuple('NamedTwo', 'one, two, three')
```

**Pull request merged**